File: haesa/0x02/2309.cpp
#include <algorithm>
#include <iostream>
using namespace std;
int main() {
  int total = 0;
  int n[9];
  for (int i = 0; i < 9; i++) {
    cin >> n[i];
    total += n[i];
  }

  sort(n, n + 9);

  int sub = total - 100;
  for (int i = 0; i < 9; i++) {
    for (int j = i + 1; j < 9; j++)
      if (n[i] + n[j] == sub) {
        for (int k = 0; k < 9; k++) {
          if (k == i || k == j) continue;
          cout << n[k] << '\n';
        }
        return 0;
      }
  }
}

File: haesa/0x02/2309.js
const fs = require('fs');
const input = fs
  .readFileSync('./input.txt')
  .toString()
  .trim()
  .split('\n')
  .map(Number);

input.sort((a, b) => a - b);

const solution = (arr) => {
  const sub = arr.reduce((acc, cur) => acc + cur, 0) - 100;

  for (let i = 0; i < 8; i++)
    for (let j = i + 1; j < 9; j++)
      if (arr[i] + arr[j] == sub) {
        arr.splice(j, 1);
        arr.splice(i, 1);
        return arr;
      }
};
console.log(solution(input).join('\n'));
